React/Modules/RCTDevSettings.m

```diff
@@ -0,0 +1,508 @@
+    if (enabled) {
+      [_bridge startProfiling];
+    } else {
+      [_bridge stopProfiling:^(NSData *logData) {
+        RCTProfileSendResult(self->_bridge, @"systrace", logData);
+      }];
+    }
+  }
+}
+
+RCT_EXPORT_METHOD(setLiveReloadEnabled:(BOOL)enabled)
+{
+  [self updateSettingWithValue:@(enabled) forKey:kRCTDevSettingLiveReloadEnabled];
+}
+
+- (void)_liveReloadSettingDidChange
+{
+  BOOL liveReloadEnabled = (self.isLiveReloadAvailable && [[self settingForKey:kRCTDevSettingLiveReloadEnabled] boolValue]);
+  if (liveReloadEnabled) {
+    [self _pollForLiveReload];
+  } else {
+    [_liveReloadUpdateTask cancel];
+    _liveReloadUpdateTask = nil;
+  }
+}
+
+RCT_EXPORT_METHOD(setHotLoadingEnabled:(BOOL)enabled)
+{
+  [self updateSettingWithValue:@(enabled) forKey:kRCTDevSettingHotLoadingEnabled];
+}
+
+- (void)_hotLoadingSettingDidChange
+{
+  BOOL hotLoadingEnabled = (self.isHotLoadingAvailable && [[self settingForKey:kRCTDevSettingHotLoadingEnabled] boolValue]);
+  if (RCTGetURLQueryParam(_bridge.bundleURL, @"hot").boolValue != hotLoadingEnabled) {
+    _bridge.bundleURL = RCTURLByReplacingQueryParam(_bridge.bundleURL, @"hot",
+                                                    hotLoadingEnabled ? @"true" : nil);
+    [_bridge reload];
+  }
+}
+
+RCT_EXPORT_METHOD(toggleElementInspector)
+{
+  BOOL value = [[self settingForKey:kRCTDevSettingIsInspectorShown] boolValue];
+  [self updateSettingWithValue:@(!value) forKey:kRCTDevSettingIsInspectorShown];
+}
+
+- (void)_inspectorSettingDidChange
+{
+  if (_isJSLoaded) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+    [self.bridge.eventDispatcher sendDeviceEventWithName:@"toggleElementInspector" body:nil];
+#pragma clang diagnostic pop
+  }
+}
+
+- (void)setExecutorClass:(Class)executorClass
+{
+  if (_executorClass != executorClass) {
+    _executorClass = executorClass;
+    _executorOverride = nil;
+  }
+  
+  if (_bridge.executorClass != executorClass) {
+    
+    // TODO (6929129): we can remove this special case test once we have better
+    // support for custom executors in the dev menu. But right now this is
+    // needed to prevent overriding a custom executor with the default if a
+    // custom executor has been set directly on the bridge
+    if (executorClass == Nil &&
+        _bridge.executorClass != objc_lookUpClass("RCTWebSocketExecutor")) {
+      return;
+    }
+    
+    _bridge.executorClass = executorClass;
+    [_bridge reload];
+  }
+}
+
+#pragma mark - internal
+
+- (void)_dataSourceDidUpdate:(NSNotification *)notification
+{
+  BOOL updateAllProperties = YES;
+  NSArray *propertiesToUpdate = @[ kRCTDevSettingProfilingEnabled ];
+  if (notification && notification.userInfo) {
+    propertiesToUpdate = [notification.userInfo objectForKey:kRCTDevSettingsUpdatedSettingsKey];
+    updateAllProperties = NO;
+  }
+  for (NSString *property in propertiesToUpdate) {
```

Comment:
This sort of interface is very weakly defined. It's very easy to break something since it's impossible to tell who may be using the API. Avoid using `NSNotification` as an interface.